mmdet/models/dense_heads/yolo_head.py

```diff
@@ -0,0 +1,652 @@
+# Copyright (c) 2019 Western Digital Corporation or its affiliates.
+
+import warnings
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.cnn import ConvModule, xavier_init
+
+from mmdet.core import force_fp32, multiclass_nms
+from ..builder import HEADS
+from .base_dense_head import BaseDenseHead
+
+_EPSILON = 1e-6
+
+
+@HEADS.register_module()
+class YOLOV3Head(BaseDenseHead):
+    """YOLOV3Head Paper link: https://arxiv.org/abs/1804.02767.
+
+    Args:
+        num_classes (int): The number of object classes (w/o background)
+        num_scales (int): The number of scales / stages.
+        num_anchors_per_scale (int): The number of anchors per scale.
+            The official implementation uses 3.
+        in_channels (List[int]): Number of input channels per scale.
+        out_channels (List[int]): The number of output channels per scale
+            before the final 1x1 layer.
+        strides (List[int]): The stride of each scale.
+            Should be in descending order.
+        anchor_base_sizes (List[List[int]]): The sizes of anchors.
+            The official implementation uses
+                [[(116, 90), (156, 198), (373, 326)],
+                [( 30, 61), ( 62,  45), ( 59, 119)],
+                [( 10, 13), ( 16,  30), ( 33,  23)]]
+        ignore_thresh (float): Set negative samples if gt-anchor iou
+            is smaller than ignore_thresh. Default: 0.5
+        one_hot_smoother (float): Set a non-zero value to enable label-smooth
+            Default: 0.
+        xy_use_logit (bool): Use log scale regression for bbox center
+            Default: False
+        balance_conf (bool): Whether to balance the confidence when calculating
+            loss. Default: False
+        conv_cfg (dict): Config dict for convolution layer. Default: None.
+        norm_cfg (dict): Dictionary to construct and config norm layer.
+            Default: dict(type='BN', requires_grad=True)
+        act_cfg (dict): Config dict for activation layer.
+            Default: dict(type='LeakyReLU', negative_slope=0.1).
+        train_cfg (dict): Training config of YOLOV3 head. Default: None.
+        test_cfg (dict): Testing config of YOLOV3 head. Default: None.
+    """
+
+    def __init__(self,
+                 num_classes,
+                 num_scales,
+                 num_anchors_per_scale,
+                 in_channels,
+                 out_channels,
+                 strides,
+                 anchor_base_sizes,
```

Comment:
Too many positional arguments.

Reply:
Solved. Now only `num_classes` and `in_channels` are positional arguments.